statsmodels/base/_screening.py

```diff
@@ -0,0 +1,323 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat May 19 15:53:21 2018
+
+Author: Josef Perktold
+License: BSD-3
+"""
+
+from collections import defaultdict
+import numpy as np
+
+from statsmodels.base._penalties import SCADSmoothed
+
+
+class ScreeningResults(object):
+    def __init__(self, screener, **kwds):
+        self.screener = screener
+        self.__dict__.update(**kwds)
+
+
+class VariableScreening(object):
+    """Ultra-high, conditional sure independence screening
+
+    This is an adjusted version of Fan's sure independence screening.
+
+    Parameters
+    ----------
+    model : instance of penalizing model
+        examples: GLMPenalized, PoissonPenalized and LogitPenalized
+    pen_weight : None or float
+        penalization weight use in SCAD penalized MLE
+    k_add : int
+        number of exog to add
+    k_max_add : int
+        maximum number of variables to add during variable addition, i.e.
+        forward selection. default is 30
+    threshold_trim : float
+        threshold for trimming parameters to zero, default is 1e-4
+    k_max_included : int
+        maximum total number of variables to include in model.
+    ranking_attr : str
+        This determines the result attribute or model method that is used for
+        the ranking of exog to include. The availability of attributes depends
+        on the model.
+        Default is 'resid_pearson', 'score_factor' can be used in GLM.
+    ranking_project : bool
+        If ranking_project is True, then the exog candidates for inclusion are
+        first projected on the already included exog before the computation
+        of the ranking measure. This brings the ranking measure closer to
+        the statistic of a score test for variable addition.
+
+    Notes
+    -----
+    Status: experimental, tested only on a limited set of models and
+    with a limited set of model options.
```

Comment:
briefly describe screening algorithm

Reply:
mention that there is no automatic tuning parameter selection yet.
mention that exog should be standardized or on similar `params` scale.
(should I add a standardize option, I guess not.)